tests/rptest/tests/topic_delete_test.py

```diff
@@ -87,30 +87,115 @@ def __init__(self, test_context):
+        self.kafka_tools.produce(topic_name,
                                  record_size=4096,
                                  num_records=2560)
 
         # Wait for segments evicted from local storage
         for i in range(0, 3):
-            wait_for_segments_removal(self.redpanda, self.topic, i, 5)
+            wait_for_segments_removal(self.redpanda, topic_name, i, 5)
 
         # Confirm objects in remote storage
-        before_objects = self.s3_client.list_objects(
-            self.si_settings.cloud_storage_bucket)
-        assert sum(1 for _ in before_objects) > 0
+        objects = self.s3_client.list_objects(
+            self.si_settings.cloud_storage_bucket, topic=topic_name)
+        assert sum(1 for _ in objects) > 0
+
+    @cluster(
+        num_nodes=3,
+        log_allow_list=[
+            'exception while executing partition operation: {type: deletion'
+        ])
+    def topic_delete_unavailable_test(self):
+        """
+        Test deleting while the S3 backend is unavailable: we should see
+        that local deletion proceeds, and remote deletion eventually
+        gives up.
+        """
+        self._populate_topic(self.topic)
+        objects_before = set(
+            self.redpanda.s3_client.list_objects(
+                self.si_settings.cloud_storage_bucket, topic=self.topic))
+        assert (len(objects_before) > 0)
+
+        with firewall_blocked(self.redpanda.nodes, self._s3_port):
+            self.kafka_tools.delete_topic(self.topic)
+
+            # From user's point of view, deletion succeeds
+            assert self.topic not in self.kafka_tools.list_topics()
+
+            # Local storage deletion should proceed even if remote can't
+            wait_until(lambda: topic_storage_purged(self.redpanda, self.topic),
+                       timeout_sec=30,
+                       backoff_sec=1)
+
+            # Erase timeout is hardcoded 60 seconds, wait long enough
+            # for it to give up.
+            time.sleep(90)
+
+            # Confirm our firewall block is really working, nothing was deleted
+            objects_after = set(
+                self.redpanda.s3_client.list_objects(
+                    self.si_settings.cloud_storage_bucket))
+            assert len(objects_after) >= len(objects_before)
+
+        # Check that after the controller backend experiences errors trying
+        # to execute partition deletion, it is still happily able to execute
+        # other operations on unrelated topics, i.e. has not stalled applying.
+        next_topic = "next_topic"
+        self.kafka_tools.create_topic(
+            TopicSpec(name=next_topic,
+                      partition_count=3,
+                      cleanup_policy=TopicSpec.CLEANUP_DELETE))
+        self._populate_topic(next_topic)
+        after_keys = set([
+            o.Key for o in self.redpanda.s3_client.list_objects(
+                self.si_settings.cloud_storage_bucket, topic=next_topic)
+        ])
```

Comment:
nit: You can pass an iterator into the `set` constructor, but it really doesn't matter here.

Reply:
cleaned this up while rebasing